### PR TITLE
Custom nxlog snippets

### PR DESCRIPTION
**Pull request by fandrei**

### CODE REVIEW

**Comment by le-bott** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.meta.logsearch.io/job/shared.releases.logsearchshipper-PRs/49/Test PASSed.

**Comment by fandrei** [comment]
@mrdavidlaing, so - a better config is finished. What do you think?

**Comment by mrdavidlaing** [comment]
@fandrei - thanks for getting this feature underway.

Could you add some tests around the nxlog config, especially for when the passed config has multiple lines.

[`src/LogsearchShipper.Core.Tests/NxLog/NxLogProcessManagerTests.cs`](
https://github.com/cityindex/LogSearchShipper/blob/master/src/LogsearchShipper.Core.Tests/NxLog/NxLogProcessManagerTests.cs) is probably the right place for the tests

**Comment by le-bott** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.meta.logsearch.io/job/shared.releases.logsearchshipper-PRs/50/Test PASSed.

**Comment by fandrei** [comment]
@mrdavidlaing, how does it look?

File: src/LogsearchShipper.Core.Tests/NxLog/NxLogProcessManagerTests.cs [hunk 1]
```diff
@@ -17,6 +17,9 @@ namespace LogSearchShipper.Core.Tests.NxLog
 	[TestFixture]
 	public class NxLogProcessManagerTests
 	{
+		private const string NxlogCustomConfigLine1 = "	Exec	$Message = substr($raw_event, 0, 1040000);";
```
Comment (mrdavidlaing) [inline src/LogsearchShipper.Core.Tests/NxLog/NxLogProcessManagerTests.cs]: @fandrei, Isn't this a duplicate of the Exec line that is already in the nxlog.conf?

It makes it unclear as to whether the `NxlogCustomConfigLine1` is checking that this always appears, or appears only once, or appears twice.

Probably better to have something else, eg: `NxlogCustomConfigLine1 = "Exec if ($raw_event =~ /DEBUG/ ) drop();`

**Comment by le-bott** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.meta.logsearch.io/job/shared.releases.logsearchshipper-PRs/51/Test PASSed.

File: src/LogsearchShipper.Core.Tests/NxLog/NxLogProcessManagerTests.cs (hunk 1 above)
Comment (fandrei) [inline src/LogsearchShipper.Core.Tests/NxLog/NxLogProcessManagerTests.cs]: @mrdavidlaing, not really a duplicate as lines are different (and we don't have to make this config a real-world config). But why not.

**Comment by mrdavidlaing** [comment]
@fandrei - I think this is ready for merging.  Click the button when you are ready

**Comment by fandrei** [comment]
@mrdavidlaing, ok

**Comment by fandrei** [comment]
@mrdavidlaing, build fails. Quite a strange error:

```
Publish artifacts to S3 Bucket bucket=ci-logsearch, file=LogSearchShipper-1.5.3-46.zip region=EU_WEST_1, upload from slave=false managed=true
FATAL: No author in changeset 7353852d565372d2d4a1aa94e32467d8c832d721
java.lang.RuntimeException: No author in changeset 7353852d565372d2d4a1aa94e32467d8c832d721
	at hudson.plugins.git.GitChangeSet.getAuthor(GitChangeSet.java:323)
	at hudson.model.AbstractBuild.getCulprits(AbstractBuild.java:348)
	at hudson.model.AbstractBuild.getCulprits(AbstractBuild.java:344)
	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:673)
	at hudson.model.Run.execute(Run.java:1731)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:88)
	at hudson.model.Executor.run(Executor.java:232)
```

**Comment by mrdavidlaing** [comment]
@fandrei - wierd - haven't seen that before.  Try push a "minor" commit to the master branch

**Comment by fandrei** [comment]
@mrdavidlaing, ok it builds now. Weird.

**Pull request merged**